### PR TITLE
feat: Add reply_to_list functionality

### PR DESCRIPTION
**Pull request by thepuzzlemaster**
`reply_to_list` exists in the sendgrid API v3, but it has never been implemented for the python api.

I added tests and a use case, but I didn't see any other docs. But this property already exists in the [official docs](https://docs.sendgrid.com/api-reference/mail-send/mail-send#multiple-reply-to-emails).

Also, I never code in python, so I may have made some mistakes here. Just kind of based this off of the existing code and some googling.

### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [x] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [x] I have read the [Contribution Guidelines](https://github.com/sendgrid/sendgrid-python/blob/main/CONTRIBUTING.md) and my PR follows them
- [x] I have titled the PR appropriately
- [x] I have updated my branch with the main branch
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added the necessary documentation about the functionality in the appropriate .md file
- [-] I have added inline documentation to the code I modified

If you have questions, please file a [support ticket](https://support.sendgrid.com).

### CODE REVIEW

**Comment by thepuzzlemaster** [comment]
Does anyone at Sendgrid maintain this repo?
@childish-sambino or @eshanholtz? I see you both have approved/merged the most recent PR's.
I don't see any information in the .md files about requesting reviewers, so not sure if I missed a step perhaps.

File: sendgrid/helpers/mail/mail.py [hunk 1]
```diff
@@ -695,6 +696,33 @@ def reply_to(self, value):
...
+            for reply in value:
+                if isinstance(reply, ReplyTo):
+                    if not isinstance(reply.email, str):
+                        raise ValueError('You must provide an email for each entry in a reply_to_list')
+                    reply = ReplyTo(reply.email, reply.name)
+                else:
+                    raise ValueError(
+                        'Please use a list of ReplyTos for a reply_to_list.'
+                    )
+                self._reply_to_list = value
```
Comment (childish-sambino) [inline sendgrid/helpers/mail/mail.py:724]: This should be outside the `for` loop.

File: sendgrid/helpers/mail/mail.py [hunk 2]
```diff
@@ -695,6 +696,33 @@ def reply_to(self, value):
...
+
+        :param value: A list of ReplyTo email addresses
+        :type value: list(ReplyTo), tuple
+        """
+        if isinstance(value, list):
+            for reply in value:
+                if isinstance(reply, ReplyTo):
+                    if not isinstance(reply.email, str):
+                        raise ValueError('You must provide an email for each entry in a reply_to_list')
+                    reply = ReplyTo(reply.email, reply.name)
```
[the remark this answers is not in the document]
Comment (thepuzzlemaster) [reply sendgrid/helpers/mail/mail.py]: Whoops. Yeah, I think you're correct. Removed it.

**Review by childish-sambino — APPROVED** [review]
LGTM

**Pull request merged**